_posts/2018-03-09-starter-kit.md

```diff
@@ -0,0 +1,176 @@
+the above requirements. It provides a simple
+[React page](https://github.com/cockpit-project/starter-kit/blob/master/src/starter-kit.jsx)
+that uses the [cockpit.file() API](http://cockpit-project.org/guide/latest/cockpit-file.html) to read `/etc/hostname`
+and show it. There is also an accompanying
+[test](https://github.com/cockpit-project/starter-kit/blob/master/test/check-starter-kit) that verifies this page. The
+other files are mostly build system boilerplate, i. e. the things you don't want to worry about as the first thing when
+you start a project.
+
+So, how to get this? Make sure you have the `npm` package installed. Then check out the repository and build it:
+```
+git clone https://github.com/cockpit-project/starter-kit.git
+cd starter-kit
+make
+```
+
+After that, install (or rather, symlink) the webpack-generated output page in `dist/` to where cockpit can see it:
+
+```
+mkdir -p ~/.local/share/cockpit
+ln -s `pwd`/dist ~/.local/share/cockpit/starter-kit
+```
+Now you should be able to log into <https://localhost:9090> and see the "Starter Kit" menu entry:
+
+![starter kit](/images/starter-kit.png)
+
+The symlink into your source code checkout is a very convenient and efficient way of development as you can just type
+`make` after changing code and directly see the effect in Cockpit after a page reload.
+
+You should now play around with this a little by hacking src/starter-kit.jsx, running `make`, and reloading the page.
+For example, try to read and show another file, run a program and show its output, or use
+[cockpit.file("/etc/hostname").watch(callback)](http://cockpit-project.org/guide/latest/cockpit-file.html)
+to react to changes of /etc/hostname and immediately update the page.
+
+## Testing
+
+Untested code is broken code. If not here and now, then in the future or some other operating system. This is why
+Cockpit has a rather complex machinery of regularly building 26 (!) VM images ranging from RHEL-7 and Fedora 27 over
+various releases of Debian and Ubuntu to OpenShift and Windows 8, and running hundreds of integration tests on each of
+them for every PR in an OpenShift cluster.
+
+Replicating this for other projects isn't easy, and this has been one, if not *the* major reason why there aren't many
+third-party Cockpit projects yet. So we now made it possible for third-party GitHub projects to use Cockpit's CI
+environment, test VM images, and (independently) Cockpit's browser test abstraction API.
+
+starter-kit uses all three of those: When you run `make check`, it will:
+
+ * build an RPM out of your current code
+ * check out cockpit's [bots/ directory](https://github.com/cockpit-project/cockpit/tree/master/bots/) that has the
+   current image symlinks and tools to download, customize and run VM images
+ * check out cockpit's [tests/common directory](https://github.com/cockpit-project/cockpit/tree/master/test/common)
```

Comment:
Ditto to above.

Reply:
Yes, here it makes sense. I'll send a starter-kit PR and then update the blog post.

Reply:
Sent https://github.com/cockpit-project/starter-kit/pull/18 and updated the blog post.